crates/biome_js_analyze/src/semantic_analyzers/nursery/no_this_in_static.rs

```diff
@@ -0,0 +1,127 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, ActionCategory};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::{JsMethodClassMember, JsStaticMemberExpression, JsSuperExpression, JsThisExpression, JsClassDeclaration};
+use biome_rowan::{declare_node_union, AstNode, AstNodeList, AstNodeExt, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Succinct description of the rule.
+    ///
+    /// Put context and details about the rule.
+    /// As a starting point, you can take the description of the corresponding _ESLint_ rule (if any).
+    ///
+    /// Try to stay consistent with the descriptions of implemented rules.
+    ///
+    /// Add a link to the corresponding ESLint rule (if any):
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/rule-name
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = 2;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// ```
+    ///
```

Comment:
Don't forget to write the description. You can use the description of https://github.com/mysticatea/eslint-plugin/blob/master/docs/rules/no-this-in-static.md.

Note that every incorrect example should be in its own code block. A code block must emit a single diagnostic.